tests/route/test_static_route.py

```diff
@@ -0,0 +1,108 @@
+import pytest
+import json
+import ipaddress
+from tests.common.utilities import wait_until
+import ptf.testutils as testutils
+import ptf.mask as mask
+import ptf.packet as packet
+import time
+
+pytestmark = [
+    pytest.mark.topology('t0'),
+    pytest.mark.device_type('vs')
+]
+
+def add_ipaddr(ptfhost, nexthop_addrs, prefix_len, nexthop_devs, ipv6=False):
+    for idx in range(len(nexthop_addrs)):
+        if ipv6:
+            ptfhost.shell("ip -6 addr add {}/{} dev eth{}".format(nexthop_addrs[idx], prefix_len, nexthop_devs[idx]), module_ignore_errors=True)
+        else:
+            ptfhost.shell("ip addr add {}/{} dev eth{}".format(nexthop_addrs[idx], prefix_len, nexthop_devs[idx]), module_ignore_errors=True)
+
+def del_ipaddr(ptfhost, nexthop_addrs, prefix_len, nexthop_devs, ipv6=False):
+    for idx in range(len(nexthop_addrs)):
+        if ipv6:
+            ptfhost.shell("ip -6 addr del {}/{} dev eth{}".format(nexthop_addrs[idx], prefix_len, nexthop_devs[idx]), module_ignore_errors=True)
+        else:
+            ptfhost.shell("ip addr del {}/{} dev eth{}".format(nexthop_addrs[idx], prefix_len, nexthop_devs[idx]), module_ignore_errors=True)
+
+def generate_and_verify_traffic(duthost, ptfadapter, ip_dst, expected_ports, ipv6=False):
+    if ipv6:
+        pkt = testutils.simple_tcpv6_packet(
+            eth_dst=duthost.facts["router_mac"],
+            eth_src=ptfadapter.dataplane.get_mac(0, 0),
+            ipv6_src='2001:db8:85a3::8a2e:370:7334',
+            ipv6_dst=ip_dst,
+            ipv6_hlim=64,
+            tcp_sport=1234,
+            tcp_dport=4321)
+    else:
+        pkt = testutils.simple_tcp_packet(
+            eth_dst=duthost.facts["router_mac"],
+            eth_src=ptfadapter.dataplane.get_mac(0, 0),
+            ip_src='1.1.1.1',
+            ip_dst=ip_dst,
+            ip_ttl=64,
+            tcp_sport=1234,
+            tcp_dport=4321)
+
+    exp_pkt = pkt.copy()
+    exp_pkt = mask.Mask(exp_pkt)
+    exp_pkt.set_do_not_care_scapy(packet.Ether, 'dst')
+    exp_pkt.set_do_not_care_scapy(packet.Ether, 'src')
+    if ipv6:
+        exp_pkt.set_do_not_care_scapy(packet.IPv6, 'hlim')
+        exp_pkt.set_do_not_care_scapy(packet.IPv6, 'chksum')
+    else:
+        exp_pkt.set_do_not_care_scapy(packet.IP, 'ttl')
+        exp_pkt.set_do_not_care_scapy(packet.IP, 'chksum')
+
+    testutils.send(ptfadapter, 5, pkt)
+    testutils.verify_packet_any_port(ptfadapter, exp_pkt, ports=expected_ports)
+
+def run_static_route_test(duthost, ptfadapter, ptfhost, prefix, nexthop_addrs, prefix_len, nexthop_devs, ipv6=False):
+    # Add ipaddresses in ptf
+    add_ipaddr(ptfhost, nexthop_addrs, prefix_len, nexthop_devs, ipv6=ipv6)
+
+    try:
+        # Add static route
+        duthost.shell("sonic-db-cli CONFIG_DB hmset 'STATIC_ROUTE|{}' nexthop {}".format(prefix, ",".join(nexthop_addrs)))
```

Comment:
May be later we can use the CLI option to add static route

Reply:
Yes, I agree. We can change to that once it is ready.